src/components/shop-dialog.tsx

```diff
@@ -27,6 +85,10 @@ export default function ShopDialog() {
   const router = useRouter();
   const locale = getLocale();
 
+  if (resources === undefined) {
```

Comment:
This actually has the side-effect to not even render the trigger (the shopping cart icon) as long as there are no resources, resulting in the icon to flicker when reloading the page.

So I would instead prefer (instead of not rendering anything at all) to conditionally render the `DialogContent` instead.